bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateProgressInfoEvent.java

```diff
@@ -103,8 +103,8 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        return String.format("The firmware update progress step for thing %s changed to %s.", thingUID,
-                progressInfo.getProgressStep().name());
+        return String.format("The firmware update progress for thing %s changed. State: %s Progress: %d.", thingUID,
+                progressInfo.getProgressStep().name(), progressInfo.getProgress());
```

Comment:
Due to the introduction of the percentage progress this could result in a NPE if no progress sequence has been defined. So we should check if progress step is not null